helix-term/src/ui/completion.rs

```diff
@@ -599,3 +552,88 @@ impl Component for Completion {
         markdown_doc.render(doc_area, surface, cx);
     }
 }
+
+/// A hook for resolving incomplete completion items.
+///
+/// From the [LSP spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_completion):
+///
+/// > If computing full completion items is expensive, servers can additionally provide a
+/// > handler for the completion item resolve request. ...
+/// > A typical use case is for example: the `textDocument/completion` request doesn't fill
+/// > in the `documentation` property for returned completion items since it is expensive
+/// > to compute. When the item is selected in the user interface then a
+/// > 'completionItem/resolve' request is sent with the selected completion item as a parameter.
+/// > The returned completion item should have the documentation property filled in.
+#[derive(Debug, Default)]
+struct ResolveHandler {
+    trigger: Option<CompletionItem>,
+    request: Option<helix_event::CancelTx>,
+}
+
+impl AsyncHook for ResolveHandler {
+    type Event = CompletionItem;
+
+    fn handle_event(
+        &mut self,
+        item: Self::Event,
+        timeout: Option<tokio::time::Instant>,
+    ) -> Option<tokio::time::Instant> {
+        if self
+            .trigger
+            .as_ref()
+            .is_some_and(|trigger| trigger == &item)
+        {
+            timeout
+        } else {
+            self.trigger = Some(item);
+            self.request = None;
+            Some(Instant::now() + Duration::from_millis(150))
```

Comment:
I understand that you don't want to introduce 10 different settings, but what about `low_latency = true` which will set all these annoying debounces to zero?